rclcpp/include/rclcpp/executor.hpp

```diff
@@ -560,6 +561,48 @@ class Executor
   virtual void
   spin_once_impl(std::chrono::nanoseconds timeout);
 
+protected:
+  // Implementation details, used by spin_until_complete and spin_for.
+  // Previouse implementation of spin_until_future_complete.
+  template<typename Condition, typename DurationT>
+  FutureReturnCode spin_until_complete_impl(Condition condition, DurationT timeout)
+  {
+    auto end_time = std::chrono::steady_clock::now();
```

Comment:
Done - I've added that after the start timestamp is calculated - since the execution of callable can take more than the single fetching status of the future. I did not want to spoil the timeout here.